topology/dynamic.go

```diff
@@ -113,66 +143,28 @@ func (t *dynamicTopology) Close() {
 
 	t.closed = true
 
-	t.w.Close()
-	t.s.Close()
+	t.watch.Close()
+	t.watchable.Close()
```

Comment:
the watch is watching the updates from config service, and then use this watchable to broadcast its values to so they are not associated.